bunsen/bunsen-core/src/main/java/com/cerner/bunsen/definitions/IdConverter.java

```diff
@@ -12,7 +12,11 @@ public IdConverter(T dataType) {
 
   @Override
   protected Object fromHapi(IPrimitiveType primitive) {
-    Preconditions.checkArgument(primitive instanceof IIdType);
+    // We do this hack to work around the issue in R4 where `id`s are of type `System.String`
+    // instead of just type `id`. Note `id` elements of FHIR types are strings and not `IIdType`!
+    if (!(primitive instanceof IIdType)) {
```

Comment:
is it possible to have an equality as opposed to a negation here? If the primitive for R4 is always System.String, maybe we can create a constant value called `R4_STRING_TYPE` and then have the if statement be if primitive is an instant of `R4_STRING_TYPE`.  WDYT?